### PR TITLE
Hyrule field

### PR DESCRIPTION
**Pull request by Ragmoa**
Closes #7

### CODE REVIEW

File: hyrule_field.logic [hunk 1]
```diff
@@ -1,11 +1,63 @@
 area HyruleField:
-  room EastHillsNorth
-  room EastHillsMid
-  room EastHillsSouth
+  room EastHillsNorth:
+    node West:
+      door West
+      ennemy WestAcroBandits
```
Comment (Ibot02) [inline hyrule_field.logic:5]: typo
```suggestion
      enemy WestAcroBandits
```

File: hyrule_field.logic [hunk 2]
```diff
@@ -1,11 +1,63 @@
...
+      door South
+    node WestFusionChest:
+      chest WestFusionChest
+    node GoldenRopeFusion:
+      ennemy GoldenRope
+    node MainEast:
+      door FarmersHouse
+      door SouthEast
+      ennemy EastAcroBandits
+      ennemy Octorocs
```
Comment (Ibot02) [inline hyrule_field.logic:15]: I see no reason to split this group of enemies. Also, same typo
```suggestion
      enemy EastAcroBanditsAndOctorocs
```

File: hyrule_field.logic [hunk 3]
```diff
@@ -1,11 +1,63 @@
...
+    node WoodsFairyLedge:
+      door NorthEast
+    node FarmersDiggingCave:
+      door DiggingCave
+    West -> WestFusionChest : flag WestFusionChest
```
Comment (Ibot02) [inline hyrule_field.logic:20]: Do we want to have a convention for fusion names? E.g. ending them all in `Fusion` like
```suggestion
    West -> WestFusionChest : flag ChestFusion
```

File: hyrule_field.logic [hunk 4]
```diff
@@ -1,11 +1,63 @@
...
+    East -> WoodsFairyLedge: item PacciCane
+    WoodsFairyLedge -> East
+    East -> FarmersDiggingCave: item MoleMitts
+    # If we use warps, you don't need mitts to exit the cave.
+    FarmersDiggingCave -> East
+  room EastHillsMid:
+    node Main:
+      door North
+      ennemy Peahats
+      ennemy SpinyBeetles
```
Comment (Ibot02) [inline hyrule_field.logic:32]: Again, same typo and no reason to split this
```suggestion
      enemy PeahatsAndSpinyBeetles
```

File: hyrule_field.logic [hunk 5]
```diff
@@ -1,11 +1,63 @@
...
+      door SouthEast
+      door SouthWest
+    node Beanstalk:
+      door Beanstalk
+    node BombableCaveEntrance:
+      door CaveEntrance
+    Main <-> Beanstalk : flag Beanstalk
+    # You don't need to bomb the entrance to exit the cave, so one-way only ? unsure
+    Main  -> BombableCaveEntrance : item BombBag
+    BombableCaveEntrance -> Main
```
Comment (Ibot02) [inline hyrule_field.logic:42]: Did you test this already (go inside, unset the flag with a memory editor, leave) or is this a guess?

File: hyrule_field.logic [hunk 6]
```diff
@@ -1,11 +1,63 @@
...
+  room EastHillsSouth:
+    node NorthWest:
+      door NorthWest
+    node West
+    node MinishHouse:
+      door MinishHouse
+    node East:
+      door NorthEast
+      door SouthEast
```
Comment (Ibot02) [inline hyrule_field.logic:51]: Probably should be just East?
```suggestion
      door East
```

File: hyrule_field.logic [hunk 7]
```diff
@@ -1,11 +1,63 @@
...
+      ennemy SpinyBeetles
+    NorthWest <-> West : item BombBag
```
Comment (Ibot02) [inline hyrule_field.logic:53]: The rock should get a flag

**Review by Ibot02 — CHANGES_REQUESTED** [review]

File: hyrule_field.logic (hunk 5 above)
Comment (Rylux) [reply hyrule_field.logic:42]: I warped myself using command box and exited the cave without having the entrance opened

File: hyrule_field.logic (hunk 2 above)
Comment (Ibot02) [reply hyrule_field.logic:15]: Enemizer can always still look inside the groups - it doesn't need to be split in the logic for that.

File: hyrule_field.logic [hunk 8]
```diff
@@ -127,16 +127,95 @@ area HyruleField:
...
     NorthFreestanding -> NorthDiggingCave: (item MoleMitts)
     NorthDiggingCave -> NorthFreestanding
-  room North
+  room North:
+    node Main:
+      door South
+      door North
+      enemy OctorcsAndCrows
+    node NWTingleTree:
+      door NWTingleTree
```
Comment (Ibot02) [inline hyrule_field.logic:145]: Doing it this way implies you can go back through the entrance if the tree is closed (you'd be able to leave, but no reenter, right?)

Comment (Rylux) [reply hyrule_field.logic:145]: Ah ?  I thought it wouldn't because of those edges
``Main -> NWTingleTree: flag NWTingleTree``
  ``NWTingleTree -> Main``

Comment (Ibot02) [reply hyrule_field.logic:146]: That just means you can't get to the entrance from anywhere else without the flag being set - if the exit and entrance are on the same node (whenever you use `door`) you are implied to be able to get back freely.

Comment (Rylux) [reply hyrule_field.logic:146]: Oh ok, so if I get this right, I use door and the same nodz only if it's the access to the door, and exit/entrance and separeted nodes when it's the door itself ?

File: hyrule_field.logic (hunk 5 above)
Comment (Ibot02) [reply hyrule_field.logic:45]: The rocks being blown up is a flag though

**Review by CandyCrystal — CHANGES_REQUESTED** [review]
Perhaps remove the empty lines (if thats how we want it to look?), also the zeffa fluteaways should be entrances, not exits

**Pull request merged**